automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/methodrepo/MethodRepoPage.scala

```diff
@@ -1,85 +1,56 @@
 package org.broadinstitute.dsde.firecloud.page.methodrepo
 
 import org.broadinstitute.dsde.firecloud.config.Config
-import org.broadinstitute.dsde.firecloud.page.components.Table
-import org.broadinstitute.dsde.firecloud.page.{AuthenticatedPage, FireCloudView, PageUtil}
+import org.broadinstitute.dsde.firecloud.component._
+import org.broadinstitute.dsde.firecloud.page.{AuthenticatedPage, OKCancelModal, PageUtil}
 import org.openqa.selenium.WebDriver
 import org.scalatest.selenium.Page
 
 class MethodRepoPage(implicit webDriver: WebDriver) extends AuthenticatedPage with Page with PageUtil[MethodRepoPage] {
 
   override val url: String = s"${Config.FireCloud.baseUrl}#methods"
 
-  override def awaitLoaded(): MethodRepoPage = {
-    ui.MethodRepoTable.awaitReady()
-    this
+  override def awaitReady(): Unit = {
+    MethodRepoTable.awaitReady()
   }
 
-  trait UI extends super.UI {
-    private val newMethodButtonQuery = testId("create-method-button")
+  private val newMethodButton = Button("create-method-button")
 
-    def clickNewMethodButton(): Unit = {
-      click on newMethodButtonQuery
-    }
+  def createNewMethod(attributes: Map[String, String]): Unit = {
+    newMethodButton.doClick()
+    val createMethodModal = await ready new CreateMethodModal()
+    createMethodModal.fillOut(attributes)
+    createMethodModal.submit()
+  }
 
-    object MethodRepoTable extends Table("methods-table") {
-      private def methodLink(namespace: String, name: String) = findInner(s"method-link-$namespace-$name")
+  object MethodRepoTable extends Table("methods-table") {
+    private def methodLink(namespace: String, name: String) = Link(s"method-link-$namespace-$name")
 
-      def hasMethod(namespace: String, name: String): Boolean = {
-        find(methodLink(namespace, name)).isDefined
-      }
+    def hasMethod(namespace: String, name: String): Boolean = {
+      val link = methodLink(namespace, name)
+      link.isVisible
+      methodLink(namespace, name).isVisible
```

Comment:
does isVisible really need to be called on both link and methodLink? if it's because of the side effects of the first.isVisible i think it should be named something else

Reply:
Nope, it was a bad merge